app.py

```diff
@@ -1,26 +1,402 @@
+
+    Used for search_mentions arg
+    """
+    try:
+        val = str(val)
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"{val} could not be parsed to a string")
+
+    if not val.startswith('@'):
+        return '@' + val
+    return val
+
+
+#########CUSTOM TYPES#########
+
+#########HELPER METHODS#########
+
+class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter, argparse.RawTextHelpFormatter):
+    """
+    Allows us to use two formatters for argparse instead of only the default 1 (hackish)
+    https://stackoverflow.com/a/18462760
+    """
+
+    def _split_lines(self, text, width):
+        # https://stackoverflow.com/a/29498894
+        # modify super to add newline between arguments
+        lines = super()._split_lines(text, width)
+        if text.endswith('\n'):  # custom newline
+            lines += ['']
+        else:
+            lines += ['']  # also add newline between different arguments
+        return lines
+
+
+def _default_dates():
+    """
+    Generates today and five days from now for use with default values of the date-argument
+    """
+    today = datetime.now().date()
+    five_days_from_now = today + timedelta(days=5)
+    # create readable format, as should be input
+    # return [today.strftime('%Y-%m-%d'), five_days_from_now.strftime('%Y-%m-%d')]
+    return [today, five_days_from_now]
+
+
+def _filter_search_values(key: str, values: list, collection: list):
+    """
+    Filters tweet data by a single key and many values.
+
+    ## Returns a list of items which [key]-collection holds any of the given values
+
+    ### Example: 
+
+    collection holds ['#trump', '#biden', '#uselection'].  
+    values hold ['#trump', '#uselection'].
+
+    This item would be returned as it matches 2 of 3 values.
+
+    >>> list (original collection)
+    """
+    return_data = []
+    for item in collection:
+        if any(val in values for val in item[key]):
+            return_data.append(item)
+    return return_data
+
+
+def _filter_data(analyzed_tweet_data: list, start_date, end_date, hashtags, mentions, urls):
+    """
+    Filters data according to given arguments.
+    Filters by date per default.
+    Other arguments default to None and will not execute unless specifically passed.
+    """
+    # filter by dates
+    filtered_data = get_tweets_in_daterange(
+        analyzed_tweet_data, start_date, end_date)
+    print("Done filtering on date...")
+    if hashtags:
+        filtered_data = _filter_search_values(
+            'hashtags', hashtags, filtered_data)
+        print(f'Done filtering on hashtags: {hashtags}')
+    if mentions:
+        filtered_data = _filter_search_values(
+            'mentions', mentions, filtered_data)
+        print(f'Done filtering on mentions: {mentions}')
+    if urls:
+        filtered_data = _filter_search_values(
+            'tweet_urls', urls, filtered_data)
+        print(f'Done filtering on urls: {urls}')
+
+    return filtered_data
```

Comment:
So this is the "Mother" method for the helper methods?

Reply:
Could say so, yeah. The flow goes:
Main  
prepare_data (which I consider the mother method, as it handles the different modules)  
_filter_data (which filters depending on arguments passed)